### ISSUE TITLE
kubespray: cert-manager too old

### ISSUE THREAD

**Issue by zonca** [zonca/jupyterhub-deploy-kubernetes-jetstream#27]
@julienchastang reported that the Kubespray HTTPS setup fails due to `cert-manager` errors.

```
I0312 19:48:11.089997       1 setup.go:173] Skipping retrying account registration as a BadRequest response was returned from the ACME server: acme: urn:ietf:params:acme:error:rateLimited: Your ACME client is too old. Please upgrade to a newer version.
```

**Comment by zonca** [comment]
## Update only cert-manager


@julienchastang a quick&dirty fix would be to just upgrade cert-manager.

See: https://github.com/zonca/jetstream_kubespray/blob/branch_v2.8.2/roles/download/defaults/main.yml#L220

You can find all the available versions at https://hub.helm.sh/charts/jetstack/cert-manager
A version which is too new would probably require different syntax of other parts of kubespray,
so I would try to use a **less old** release, maybe `v0.9.1`?

**Comment by zonca** [comment]
It looks like we are hitting this problem: https://community.letsencrypt.org/t/blocking-old-cert-manager-versions/98753

There they say after 3 months they could decide to block, if that is the case you should try v0.12.0 or a later version.

**Comment by zonca** [comment]
@julienchastang have you considered to switch to the Magnum-based deployment instead? Is there anything that is holding you off?
Magnum deploys a lot quicker.

**Comment by julienchastang** [comment]
Thanks for the quick response. My immediate concern is online class support and I have not had time to familiarize myself with the magnum deploy (even if it is simpler). I need a stop gap measure in the interim while I adjust to the magnum workflow. (I was caught off guard by the rapidity of the cov-19 situation.) I may simply re-use another one of our JupyterHubs which uses traditional certs. 

Or I may use the interim solution you propose here. When you refer to `v0.12.0 ` are you referring to the certmanager version referenced here: https://github.com/zonca/jetstream_kubespray/blob/branch_v2.8.2/roles/download/defaults/main.yml#L220?

Thanks again.

**Comment by julienchastang** [comment]
It seems like `https_issuer.yml` and the `https_certificate.yml` would need to be updated too, correct?

**Comment by zonca** [comment]
> Or I may use the interim solution you propose here. When you refer to `v0.12.0 ` are you referring to the certmanager version referenced here: https://github.com/zonca/jetstream_kubespray/blob/branch_v2.8.2/roles/download/defaults/main.yml#L220?

yes, right 

`https_issue.yml` and `https_certificate.yml` I guess you are right as well,
Actually I think I can test this and update the instructions.
Give me half-an-hour.

**Comment by zonca** [comment]
ok, here it is with `0.14`: https://zonca.dev/2020/03/setup-https-kubernetes-letsencrypt.html

**Comment by zonca** [comment]
ok, I'll close this, please reopen if you hit any trouble

**Comment by julienchastang** [comment]
Cool. Plus, I also see you adjusted the zonca/jupyterhub-deploy-kubernetes-jetstream repo. What about the zonca/jetstream_kubespray repo? Does that need to reference a new version of the certmanager? Or should I just do the manual adjustments you suggested earlier? Thanks again.

**Comment by zonca** [comment]
@julienchastang oh yes, I tested this on the Magnum deployment, easiest is to deploy with kubespray, then delete cert-manager, and follow my new tutorial

**Comment by julienchastang** [comment]
OK, so do the kubespray deploy as usual, then delete the certmanager with `kubectl delete namespace certmanager` and follow the new instructions? Are any `helm` commands required to remove the certmanager (e.g., `helm delete certmanager --purge`). I'm extrapolating from how I delete `jhub` namespaces.

**Comment by zonca** [comment]
actually, I disabled cert-manager in the kubespray recipe, see the commit reference above here

**Comment by zonca** [comment]
do you already have a kubespray deployment running?

**Comment by julienchastang** [comment]
yes, but that is OK, I don't mind starting from scratch (if that is why you are asking).

**Comment by zonca** [comment]
ok, but before starting from scratch you could try to just delete the namespace, that deletes all the resources in that namespace so it should be enough

**Comment by julienchastang** [comment]
Also it looks like only the `https_issuer` step is required. The previous `https_certificate` is no longer needed:

```
kubectl create -f setup_https_kubespray/https_certificate.yml
```

(Also, I know the new directory for the issuer is `setup_https/`.)

**Comment by zonca** [comment]
Correct, it is automatic

**Comment by julienchastang** [comment]
This seems to all work as advertised as far as getting the cert, but then when trying to start and access the JupyterHub in the usual manner:

![image](https://user-images.githubusercontent.com/229955/76659014-02a40180-653b-11ea-91d2-61e8eba318b5.png)

```
hub:
  cookieSecret: "xxx"

proxy:
  secretToken: "xxx"

ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: "nginx"
    cert-manager.io/issuer: "letsencrypt"
  hosts:
      - js-170-84.jetstream-cloud.org
  tls:
      - hosts:
         - js-170-84.jetstream-cloud.org
        secretName: certmanager-tls-jupyterhub

singleuser:
  image:
    name: jupyter/minimal-notebook
    tag: latest
```

**Comment by julienchastang** [comment]
Permanently "pending": 

![image](https://user-images.githubusercontent.com/229955/76659729-c4a7dd00-653c-11ea-81b2-aba627086178.png)

**Comment by zonca** [comment]
Need to check the logs, this works fine in my deployment

**Comment by julienchastang** [comment]
Have you ever seen:

```
I0314 02:57:24.026417       9 event.go:221] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"jhub", Name:"jupyterhub", UID:"87c41f2c-659f-11ea-ada6-fa163e7f3e0f", APIVersion:"extensions/v1beta1", ResourceVersion:"11244", FieldPath:""}): type: 'Normal' reason: 'CREATE' Ingress jhub/jupyterhub
W0314 02:57:27.049773       9 controller.go:826] Service "jhub/proxy-public" does not have any active Endpoint.
I0314 02:57:27.050012       9 controller.go:172] Configuration changes detected, backend reload required.
I0314 02:57:27.340869       9 controller.go:190] Backend successfully reloaded.
[14/Mar/2020:02:57:27 +0000]TCP200000.000
W0314 02:57:30.383187       9 controller.go:826] Service "jhub/proxy-public" does not have any active Endpoint.
I0314 02:57:42.172939       9 status.go:388] updating Ingress jhub/jupyterhub status to [{149.165.170.84 }]
I0314 02:57:42.178509       9 event.go:221] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"jhub", Name:"jupyterhub", UID:"87c41f2c-659f-11ea-ada6-fa163e7f3e0f", APIVersion:"extensions/v1beta1", ResourceVersion:"11339", FieldPath:""}): type: 'Normal' reason: 'UPDATE' Ingress jhub/jupyterhub
W0314 02:57:42.180264       9 controller.go:826] Service "jhub/proxy-public" does not have any active Endpoint.
```

**Comment by zonca** [comment]
No, does it only happen with cert-manager? Was it working before?

**Comment by julienchastang** [comment]
Right now, I'm just trying to get the most basic setup going withouts certs. So

```
hub:
  cookieSecret: "xxx"

proxy:
  secretToken: "xxx"

ingress:
  enabled: true
  hosts:
    - js-170-84.jetstream-cloud.org
```

I just don't get it because apart from the cert-manager we did not change anything in `jupyterhub-deploy-kubernetes-jetstream` or in `jetstream_kubespray`. I feel like I am missing something obvious.

**Comment by julienchastang** [comment]
I was able to get things working again by going back to a previous version of `jupyterhub-deploy-kubernetes-jetstream`. My present working hypothesis is the problem lies there. I'll try to give more details over the weekend.

**Comment by zonca** [comment]
ok, thanks, I might have broken something working on the magnum deployment

**Comment by julienchastang** [comment]
I can confirm that going back to a previous version of `jupyterhub-deploy-kubernetes-jetstream`, for example `a0f29fa4409c27e2b762291e8183abc62072633b` from Dec 18, can make things work again. I am not going to pursue this any further since I have something that works and this is all stopgap anyway until I transition to magnum. Thanks again for your help here!
